### PR TITLE
Starting the Language Design Choices

### PR DESCRIPTION
**Pull request by mariari**

### CODE REVIEW

File: doc/Language/Proof-Search.org [hunk 1]
```diff
@@ -0,0 +1,47 @@
+* Proof Searching of Juvix
+- Data Structures in Juvix should provide complimentary proofs that
+  are relevant to the users.
+** The Argument for Implicit Refinements Over Indexed Types For Structures
```
Comment (cwgoes) [inline doc/Language/Proof-Search.org]: This looks like a bit about what the standard library should provide? I think we might want a file specifically for the stdlib.

File: doc/Language/Interactive.org [hunk 2]
```diff
@@ -0,0 +1,3 @@
+* Interaction of the Juvix Programming Language
+In many respects Juvix should
```
Comment (cwgoes) [inline doc/Language/Interactive.org]: should what?

What a cliffhanger!

File: doc/Language/Proof-Search.org (hunk 1 above)
Comment (mariari) [reply doc/Language/Proof-Search.org]: We probably should have one, however I think this should stay here.

File: doc/Language/Interactive.org (hunk 2 above)
Comment (mariari) [reply doc/Language/Interactive.org]: That is for you to fill in, and for me to find out!

**Pull request merged**